src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -2357,3 +2366,1735 @@ IsForeignTable(Oid relationId)
+		 * currently have, if any we know that the message has changed and we update the
+		 * message
+		 */
+		bool updateMessage = false;
+		if (isnull[Anum_pg_dist_background_task_message - 1])
+		{
+			updateMessage = true;
+		}
+		else
+		{
+			text *oldMessageText =
+				DatumGetTextP(values[Anum_pg_dist_background_task_message - 1]);
+			char *oldMessage = text_to_cstring(oldMessageText);
+			if (strcmp(oldMessage, task->message) != 0)
+			{
+				updateMessage = true;
+			}
+		}
+
+		if (updateMessage)
+		{
+			values[Anum_pg_dist_background_task_message - 1] =
+				CStringGetTextDatum(task->message);
+			isnull[Anum_pg_dist_background_task_message - 1] = false;
+			replace[Anum_pg_dist_background_task_message - 1] = true;
+			updated |= true;
+		}
+	}
+	else
+	{
+		updated |= SetFieldNull(Anum_pg_dist_background_task_message, values, isnull,
+								replace);
+	}
+
+	if (updated)
+	{
+		heapTuple = heap_modify_tuple(heapTuple, tupleDescriptor, values, isnull,
+									  replace);
+
+		CatalogTupleUpdate(pgDistBackgroundTasks, &heapTuple->t_self, heapTuple);
+
+		CommandCounterIncrement();
+	}
+
+	systable_endscan(scanDescriptor);
+	table_close(pgDistBackgroundTasks, NoLock);
+}
+
+
+/*
+ * GetDependantTasks returns a list of taskId's containing all tasks depending on the task
+ * passed via its arguments.
+ *
+ * Becasue tasks are int64 we allocate and return a List of int64 pointers.
+ */
+static List *
+GetDependantTasks(int64 jobId, int64 taskId)
+{
+	Relation pgDistBackgroundTasksDepends =
+		table_open(DistBackgroundTaskDependRelationId(), RowExclusiveLock);
+
+	ScanKeyData scanKey[2] = { 0 };
+	const bool indexOK = true;
+
+	/* pg_dist_background_task_depend.job_id = $jobId */
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_background_task_depend_job_id,
+				BTEqualStrategyNumber, F_INT8EQ, Int64GetDatum(jobId));
+
+	/* pg_dist_background_task_depend.depends_on = $taskId */
+	ScanKeyInit(&scanKey[1], Anum_pg_dist_background_task_depend_depends_on,
+				BTEqualStrategyNumber, F_INT8EQ, Int64GetDatum(taskId));
+
+	SysScanDesc scanDescriptor =
+		systable_beginscan(pgDistBackgroundTasksDepends,
+						   DistBackgroundTaskDependDependsOnIndexId(),
+						   indexOK,
+						   NULL, lengthof(scanKey), scanKey);
+
+	List *dependantTasks = NIL;
+	HeapTuple heapTuple = NULL;
+	while (HeapTupleIsValid(heapTuple = systable_getnext(scanDescriptor)))
+	{
+		Form_pg_dist_background_task_depend depend =
+			(Form_pg_dist_background_task_depend) GETSTRUCT(heapTuple);
+
+		int64 *dTaskId = palloc0(sizeof(int64));
+		*dTaskId = depend->task_id;
+
+		dependantTasks = lappend(dependantTasks, dTaskId);
+	}
+
+	systable_endscan(scanDescriptor);
+	table_close(pgDistBackgroundTasksDepends, NoLock);
+
+	return dependantTasks;
+}
+
+
+/*
+ * CancelTasksForJob cancels all tasks associated with a job that are not currently
+ * running and are not already in their termnial state. Canceling these tasks consist of
```

Comment:
```suggestion
 * running and are not already in their terminal state. Canceling these tasks consist of
```

Reply:
lol, that exact spelling mistake was 4 times in the codebase :')